test/dynamo/test_repros.py

```diff
@@ -2226,6 +2228,45 @@ def f(x):
         )
         self.assertEqual(gm(inp).shape, f(inp).shape)
 
+    def test_torch_out_global(self):
+        def fn(x):
+            torch.full(size=global_y.shape, fill_value=1.0, out=global_y)
+            return global_y + x
+
+        cnt = torch._dynamo.testing.CompileCounter()
+        opt_fn = torch._dynamo.optimize(cnt, nopython=True)(fn)
+        opt_fn(torch.randn(2, 2))
+        # Check that the global_y is updated
+        self.assertTrue(same(global_y, torch.ones(2)))
```

Comment:
You need to reset global_y before the test, because the test runner may run your test multiple times and after the first run global y will be spuriously updated